### ISSUE TITLE
improve the output format for RESTler bugs

### ISSUE THREAD

**Issue by marina-p** [microsoft/restler-fuzzer#537]
The following improvements should be made to the RESTler results format so results are more easily consumed by tools in a CI/CD test pipeline (e.g. for automatic bug filing):

A single json file that includes the following:
- full bug details: the current contents of bug_buckets.txt and full payloads (similar to the 'request_sample' in speccov.json including the sequence information).  The endpoint and method should correspond to the ones from the specification, so they can be shown in a results UI consistent with the spec (these are available as the request IDs in the grammar).
- any global settings that should be re-used for the repro.  For example, 'use_ssl' can be specified on the command line or in the engine settings - this setting needs to be included in the results.
- The name and description of each checker that found the bug (so these can be directly included when a bug is auto-filed).

**Comment by corygehr** [comment]
I might suggest directly generating a SARIF file as the result instead of adjusting what you already have, assuming that was your eventual plan (no need to change the format if others are taking dependencies on the current format). I think it solves most of these notes, especially for consumption by CI/CD Pipeline tools.

**Comment by marina-p** [comment]
There are other use cases for which we need the above results file, to be parsed by other tools which are not SARIF-aware, and it will be faster to implement.  We'll definitely consider adding the ability for RESTler to also have an option to generate a result report in SARIF (I'll open up a separate issue for this).

**Comment by corygehr** [comment]
Fair enough. What you've written above would fix what we were working around in my project. Thank you!
